merkle/merkle_path.go

```diff
@@ -32,13 +32,13 @@ const vvLevel = 4
 // NodeFetch bundles a nodeID with additional information on how to use the node to construct the
 // correct proof.
 type NodeFetch struct {
-	NodeID storage.NodeID
+	ID     compact.NodeID
 	Rehash bool
 }
 
 // Equivalent return true iff the other represents the same rehash state and NodeID as the other.
 func (n NodeFetch) Equivalent(other NodeFetch) bool {
-	return n.Rehash == other.Rehash && n.NodeID.Equivalent(other.NodeID)
+	return n.Rehash == other.Rehash && n.ID == other.ID
```

Comment:
`Equivalent` in `storage.NodeID` is a deep compare on the path prefix. There's no similar concept in `compact.NodeID` so I think this version of `Equivalent` should be renamed to avoid confusion.

Alternatively, as it's only used in tests so it maybe can be deleted and the structs compared directly in the test.